docs/NotesOnConnectionAccountStates.md

```diff
@@ -0,0 +1,97 @@
+# Notes On Domain Connection and Account Login for Vircadia Web
+
+[Vircadia Web] has two connections to manage: the connection to the domain-server
+and the connection to the metaverse-server.
+The domain-server connection has to do with the display and interaction with
+the virtual world scene while the connection to the metaverse-server has to do
+with the account, access tokens, and grid management.
+
+This note discusses the states for these two connections and how they interact.
+
+# domain-server Connection
+
+The connection to the domain-server is implemented by the [Vircadia Web SDK] library.
+The connection is a little complex in that a connection to a [Vircadia Server]
+includes connections to multiple assignment-clients. The connections to the
+assignment-clients are optional (like connecting to the message assignment-client
+for text chat but not connecting to the audio assignment-client).
+
+So the connection to the domain-service can be considered as a main connection
+and then several associated sub-service connections.
+
+The persisted domain-server state will be saved as the connection URL to the domain-server,
+which sub-services are connected, and optionally an "explore" connection description
+which would give name and thumbnail information.
+
+The domain-server states will be:
+
+* disconnected
+* connecting (requires connection URL)
+* connected
+* disconnecting
+* error
```

Comment:
The Web SDK has states:
![image](https://user-images.githubusercontent.com/7455448/133695618-5ed2bc5f-eb61-4797-849d-5e383e79164a.png)

Reply:
How do you use "refused"? It's just a transition state out of "connecting" that's different than "error"? I would think that "error" would cover that state and other conditions ("proxy refused", "TLS init failure", "no route to host", ...) that the caller would also need to know.

Reply:
I've been using "uninitialized" as the state structure initial state before parameters are either supplied by the user or persistent storage. Operationally, it's not entered but only left and is mostly for the developer. In vircadia-web, there can be a fairly long time between code loading and when all the sub-systems are initialized.

Reply:
After working with my code for a bit, "uninitialized" is not needed and I will remove it.

Reply:
`REFUSED`: You made a network connection to the domain server but you weren't permitted to connect to the domain, e.g., protocol mismatch, too many users, you don't have permissions, ....